Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java

```diff
@@ -274,7 +357,56 @@ public boolean onCreateOptionsMenu(Menu menu) {
 	@Override
 	public boolean onPrepareOptionsMenu(Menu menu) {
 		super.onPrepareOptionsMenu(menu);
-		menu.findItem(R.id.action_add_item).setEnabled(editable);
+		boolean UserCheck = user.getName().contentEquals(claim.getUser().getName());//SHOULD BE ID USING NAME FOR TESTING
+
+		if (status ==Status.APPROVED){
+			menu.findItem(R.id.action_add_item).setEnabled(false);
+			menu.findItem(R.id.action_mark_submitted).setEnabled(false);
+			menu.findItem(R.id.action_mark_approved).setEnabled(false);
+			menu.findItem(R.id.action_mark_returned).setEnabled(false);
+		}
+		if(status == Status.RETURNED){
+			if(UserCheck){
+				menu.findItem(R.id.action_add_item).setEnabled(true);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(true);
+				menu.findItem(R.id.action_mark_approved).setEnabled(false);
+				menu.findItem(R.id.action_mark_returned).setEnabled(false);
+			}
+			else{
+				menu.findItem(R.id.action_add_item).setEnabled(false);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(false);
+				menu.findItem(R.id.action_mark_approved).setEnabled(false);
+				menu.findItem(R.id.action_mark_returned).setEnabled(false);
+			}
+		}
+		if(status==Status.SUBMITTED){
+			if(UserCheck){
+				menu.findItem(R.id.action_add_item).setEnabled(false);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(false);
+				menu.findItem(R.id.action_mark_approved).setEnabled(false);
+				menu.findItem(R.id.action_mark_returned).setEnabled(false);
+			}
+			else{
+				menu.findItem(R.id.action_add_item).setEnabled(false);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(false);
+				menu.findItem(R.id.action_mark_approved).setEnabled(true);
+				menu.findItem(R.id.action_mark_returned).setEnabled(true);
+			}
+		}
+		if(status== Status.IN_PROGRESS){
+			if(UserCheck){
+				menu.findItem(R.id.action_add_item).setEnabled(true);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(true);
+				menu.findItem(R.id.action_mark_approved).setEnabled(false);
```

Comment:
For example, these two lines:

```
menu.findItem(R.id.action_mark_approved).setEnabled(false);
menu.findItem(R.id.action_mark_returned).setEnabled(false);
```

Are the same whether `UserCheck` is false or true when the status is `IN_PROGRESS`. So you could move these lines out to the bottom of the parent `if` block to avoid duplication. Same goes for a few other ones up there.